### PR TITLE
Developer docs about existing subtrees.

### PR DESCRIPTION
**Pull request by gmaxwell**
Add some notes about the workflow for changes to code that exists in subtrees.

### CODE REVIEW

**Comment by laanwj** [comment]
ACK

File: doc/developer-notes.md [hunk 1]
```diff
@@ -408,6 +408,34 @@ GUI
...
+be sent upstream but bugfixes may also be prudent to PR against Bitcoin Core so that they can be integrated
+quickly.  Cosmetic changes should be purely taken upstream.
+
+Current subtrees include:
+
+- src/leveldb
+  - Upstream at https://github.com/google/leveldb ; Maintained by Google, but open important PRs to Core to avoid delay
+
+- src/libsecp256k1
+  - Upstream at https://github.com/bitcoin-core/secp256k1/ ; actively maintaned by Core contributors.
```
Comment (paveljanik) [inline doc/developer-notes.md]: maintaned -> maintained

File: doc/developer-notes.md [hunk 2]
```diff
@@ -408,6 +408,34 @@ GUI
     should not interact with the user. That's where View classes come in. The converse also
     holds: try to not directly access core data structures from Views.
 
+Subtrees
+----------
+
+Several parts of the repository are subtrees of software maintained elsewhere.
+
+ Some of these are maintained by active developers of Bitcoin Core, in which case changes should probably go
```
Comment (paveljanik) [inline doc/developer-notes.md]: Why SPC at the beginning? To mark paragraph?

Two spaces?

**Comment by paveljanik** [comment]
Concept ACK. This is needed.

Travis failure in `sendheaders.py` unrelated but strange 8)

File: doc/developer-notes.md [hunk 3]
```diff
@@ -408,6 +408,34 @@ GUI
...
+directly upstream without being PRed directly against the project.  They will be merged back in the next
+subtree merge.
+
+ Others are external projects without a tight relationship with our project.  Changes to these should also
+be sent upstream but bugfixes may also be prudent to PR against Bitcoin Core so that they can be integrated
+quickly.  Cosmetic changes should be purely taken upstream.
+
+Current subtrees include:
```
Comment (MarcoFalke) [inline doc/developer-notes.md]: Maybe link to https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#git-subtree-checksh, so we don't have to keep track to update 2 places whenever something changes?

**Review by maflcko — COMMENTED** [review]
Concept ACK

**Comment by fanquake** [comment]
ACK

**Comment by jonasschnelli** [comment]
Nice.
Agree with @MarcoFalke to add a link to https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#git-subtree-checksh somewhere.

**Comment by gmaxwell** [comment]
Updated for nits.

**Pull request merged**